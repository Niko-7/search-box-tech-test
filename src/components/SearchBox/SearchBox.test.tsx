/*eslint-disable */
import { cleanup, render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { SearchBox } from ".";
const { axe } = require("jest-axe");
import "jest-axe/extend-expect";

afterEach(() => {
  cleanup();
});

describe("Snapshot tests", () => {
  it("should match the snapshot of the SearchBox component", () => {
    const tree = renderer.create(<SearchBox />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("accesibility tests", () => {
  it("should have no accesibility errors on SearchBox component", async () => {
    const { container } = render(<SearchBox />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
