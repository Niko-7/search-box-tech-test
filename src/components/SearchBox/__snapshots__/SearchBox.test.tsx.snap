// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests should match the snapshot of the SearchBox component 1`] = `
<div
  className="searchbox"
>
  <div
    className="searchbox-main-container"
  >
    <div
      className="searchwidget__input-container"
    >
      <span
        aria-hidden="true"
        className="icon"
        data-testid="icon-container"
        role="presentation"
      >
        <svg
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            d="M17.464 6.56a8.313 8.313 0 1 1-15.302 6.504A8.313 8.313 0 0 1 17.464 6.56zm1.38-.586C16.724.986 10.963-1.339 5.974.781.988 2.9-1.337 8.662.783 13.65c2.12 4.987 7.881 7.312 12.87 5.192 4.987-2.12 7.312-7.881 5.192-12.87zM15.691 16.75l7.029 7.03a.75.75 0 0 0 1.06-1.06l-7.029-7.03a.75.75 0 0 0-1.06 1.06z"
          />
        </svg>
      </span>
      <input
        aria-label="Pick-up Location"
        autoComplete="off"
        className="search-input"
        onChange={[Function]}
        placeholder="Pick-up Location"
        type="text"
      />
      <noscript />
    </div>
  </div>
</div>
`;
